docs/jax-101/02-jitting.ipynb

```diff
@@ -0,0 +1,570 @@
+    "\n",
+    "Note: the python `print()` function is not pure: the text output is a side-effect of the function. Therefore, any `print()` calls will only happen during tracing, and will not appear in the jaxpr:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "id": "JxV2p7e2RawC",
+    "outputId": "af7472de-df57-412f-83cb-e468c4b02d64"
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "printed x: Traced<ShapedArray(float32[], weak_type=True)>with<DynamicJaxprTrace(level=1/0)>\n",
+      "{ lambda  ; a.\n",
+      "  let b = log a\n",
+      "      c = log 2.0\n",
+      "      d = div b c\n",
+      "  in (d,) }\n"
+     ]
+    }
+   ],
+   "source": [
+    "def log2_with_print(x):\n",
+    "  print(\"printed x:\", x)\n",
+    "  ln_x = jnp.log(x)\n",
+    "  ln_2 = jnp.log(2.0)\n",
+    "  return ln_x / ln_2\n",
+    "\n",
+    "print(jax.make_jaxpr(log2_with_print)(3.))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "f6W_YYwRRwGp"
+   },
+   "source": [
+    "See how the printed `x` is a `Traced` object? That's the JAX internals at work.\n",
+    "\n",
+    "The fact that the python code runs at least once is strictly an implementation detail, and so shouldn't be relied upon. However, it's useful to understand as you can use it when debugging to print out intermediate values of a computation."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "PgVqi6NlRdWZ"
+   },
+   "source": [
+    "A key thing to understand is that jaxpr captures the function as executed on the parameters given to it. For example, if we have a conditional, jaxpr will only know about the branch we take:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "id": "hn0CuphEZKZm",
+    "outputId": "669c5880-790d-4e5c-b635-f3acc21a0e02"
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{ lambda  ; a.\n",
+      "  let \n",
+      "  in (a,) }\n"
+     ]
+    }
+   ],
+   "source": [
+    "def log2_if_rank_2(x):\n",
+    "  if x.ndim == 2:\n",
+    "    ln_x = jnp.log(x)\n",
+    "    ln_2 = jnp.log(2.0)\n",
+    "    return ln_x / ln_2\n",
+    "  else:\n",
+    "    return x\n",
+    "\n",
+    "print(jax.make_jaxpr(log2_if_rank_2)(jax.numpy.array([1, 2, 3])))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "Qp3WhqaqvHyD"
+   },
+   "source": [
+    "## Jitting a function\n",
```

Comment:
Suggestion: write "jit" as "JIT" (similar to https://jax.readthedocs.io/en/latest/notebooks/thinking_in_jax.html#to-jit-or-not-to-jit)


```suggestion
    "## JITting a function\n",
```

(as in Just-In-Time-ing... 🤔 )